### PR TITLE
FEATURE: assign device owner from Discourse group alert PM.

### PR DESCRIPTION
**Pull request by vinothkannans**
Screenshot 2022-02-11 at 1 37 01 PM

### CODE REVIEW

**Review by techAPJ — APPROVED** [review]

File: spec/requests/devices_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      group.add(kolide_admin)
+      sign_in(kolide_admin)
+
+      put "/kolide/devices/#{device.id}/assign.json", params: {
+        user_id: user.id
+      }
+
+      expect(response.status).to eq(200)
+      expect(device.reload.user_id).to eq(user.id)
+    end
```
Comment (nattsw) [inline spec/requests/devices_controller_spec.rb:35]: Would you perhaps like to test line 13 and 21 of the implementation too?

File: assets/javascripts/discourse/initializers/extend-for-kolide.js.es6 [hunk 2]
```diff
@@ -1,9 +1,35 @@
...
+      data: { user_id: userId },
+    })
+      .then(() => {
+        bootbox.alert("Success");
+      })
+      .catch(popupAjaxError);
+
+    return false;
+  }
+
```
Comment (nattsw) [inline assets/javascripts/discourse/initializers/extend-for-kolide.js.es6:32]: Would be nice if there could be some component / acceptance tests. As I've not seen it before, I can't visualise how these things get triggered.

Comment (nattsw) [reply assets/javascripts/discourse/initializers/extend-for-kolide.js.es6:32]: One thing which would also be useful is something like [tgx's PR](https://github.com/discourse/discourse-question-answer/pull/18) where he inserts certain screenshots to allow the reviewer to understand what the code changes mean

File: assets/javascripts/lib/discourse-markdown/kolide-assign.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    "a[data-device]",
+  ]);
+
+  helper.registerPlugin((md) => {
+    const rule = {
+      matcher: /\[kolide-assign user\=(.+?) device\=(.+?)\]/,
+      onMatch: addAssignButton,
+    };
+
+    md.core.textPostProcess.ruler.push("kolide-assign", rule);
```
Comment (nattsw) [inline assets/javascripts/lib/discourse-markdown/kolide-assign.js:47]: Some tests to validate this rule would be nice.

File: assets/javascripts/discourse/initializers/extend-for-kolide.js.es6 (hunk 2 above)
Comment (vinothkannans) [reply assets/javascripts/discourse/initializers/extend-for-kolide.js.es6:32]: I added a screenshot in description 👍

File: assets/javascripts/lib/discourse-markdown/kolide-assign.js (hunk 3 above)
Comment (vinothkannans) [reply assets/javascripts/lib/discourse-markdown/kolide-assign.js:47]: We already have a test [here](https://github.com/discourse/discourse-kolide/pull/33/files#diff-3663acc8c1783bbc5a0d1f5674dc23c146009d547d39d0b25859a2dcd2a8ab7bR10-R11).

**Comment by vinothkannans** [comment]
@nattsw sorry for the big delay. I did the changes. Also, I changed some of the functionalities [here](https://github.com/discourse/discourse-kolide/pull/33/commits/ba5c113800d72abf78e33bcb1333f1ab57a3c149). Can you please check it now?

**Review by nattsw — APPROVED** [review]

**Pull request merged**